src/components/Main/Main.js

```diff
@@ -16,6 +18,9 @@ export default function Main() {
     error
   } = useCountries();
 
+  if (loading) {
+    return <div className="loader"></div>;}
```

Comment:
The way you've set the loading state looks great, but I notice nothing seems to change on loading. I think this may be because, if loading is true, you are just rendering an empty div. I would try adding some content in this div (the string "Loading...", or whatever) and see if that does it.